psh/sysexec/sysexec.c

```diff
@@ -150,15 +157,57 @@ static int psh_sysexec(int argc, char **argv)
 		return -EINVAL;
 	}
 
-	pid = spawnSyspage(codeMapName, dataMapName, progName, &argv[optind]);
-	if (pid > 0) {
+	pid_t pid;
+	if (background) {
+		pid = vfork();
+	}
+	else {
+		pid = spawnSyspage(codeMapName, dataMapName, progName, &argv[optind]);
+	}
+
+	/* Forked path */
+	if (pid == 0) {
+		/* Create a new SID for the child process */
+		int sid = setsid();
+		if (sid < 0) {
+			fprintf(stderr, "psh: setsid failed: %s\n", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+
+		/* Make new process write to klog instead of our console */
+		int err = -1;
+		int klog = open(_PATH_KLOG, O_WRONLY);
+		if (klog >= 0) {
+			err = dup2(klog, STDOUT_FILENO);
+			if (err == 0) {
+				err = dup2(klog, STDERR_FILENO);
+			}
+			close(klog);
+		}
```

Comment:
I think the process should keep writing to console and if the user should explicitly redirect output. In current impementation `sysexec -d hello > file` doesn't work despite being valid command.

Reply:
that's right `&> /dev/null` for no logs at all nither klog and console, btw. I wonder if maybe `sysexec -d hello &> /dev/klog` should be the way to go to redirect all stdio & stderr to klog ?

Reply:
Ok, removed redirection as a lesser evil here. Will wreck havoc if new process tries to read from stdin though

Reply:
Can't we keep stdint closed?

Reply:
Added `-i` option